File: src/main/java/com/example/the_machine/service/mapping/ThreadMapper.java
package com.example.the_machine.service.mapping;

import com.example.the_machine.domain.ThreadEntity;
import com.example.the_machine.dto.ThreadDto;
import org.mapstruct.Mapper;

@Mapper(config = GlobalMapperConfig.class)
public interface ThreadMapper {

  ThreadDto toDto(ThreadEntity entity);

  ThreadEntity toEntity(ThreadDto dto);
}

File: src/main/kotlin/com/example/the_machine/service/mapping/ThreadMapper.kt
package com.example.the_machine.service.mapping

import com.example.the_machine.domain.ThreadEntity
import com.example.the_machine.dto.ThreadDto
import org.mapstruct.Mapper

@Mapper(config = GlobalMapperConfig::class)
interface ThreadMapper {

  fun toDto(entity: ThreadEntity): ThreadDto

  fun toEntity(dto: ThreadDto): ThreadEntity
}